File: recommendation_system/models/director.py
from datetime import datetime

from models.person import Person


class Director(Person):
    def __init__(self, id: int, first_name: str, last_name: str, date_of_birth: datetime, biography: str,
                 nationality: str, filmography: list, style: list):
        super().__init__(id, first_name, last_name, date_of_birth, biography, nationality)
        self._filmography = filmography
        self._style = style

    def set_filmography(self, filmography: list):
        self._filmography = filmography

    def get_filmography(self):
        return self._filmography

    def set_style(self, style: list):
        self._style = style

    def get_style(self):
        return self._style

File: movie/src/main/java/com/example/movie/model/Director.java
package com.example.movie.model;

import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.OneToMany;

import java.util.Date;
import java.util.List;
import java.util.UUID;

@Entity
public class Director extends Person {
    @OneToMany
    protected List<Movie> filmography;
    @OneToMany
    @Enumerated(EnumType.ORDINAL)
    protected List<Genre> style;

    public Director(String firstName, String lastName, Date dateOfBirth, String nationality, String biography, UUID profileImage, List<Movie> filmography, List<Genre> style) {
        super(firstName, lastName, dateOfBirth, nationality, biography, profileImage);
        this.filmography = filmography;
        this.style = style;
    }

    public Director() {

    }

    public List<Movie> getFilmography() {
        return filmography;
    }

    public void setFilmography(List<Movie> filmography) {
        this.filmography = filmography;
    }

    public List<Genre> getStyle() {
        return style;
    }

    public void setStyle(List<Genre> style) {
        this.style = style;
    }

    public void addStyle(Genre style) {
        this.style.add(style);
    }

    public void deleteStyle(Genre style) {
        this.style.remove(style);
    }

    @Override
    public String toString() {
        return "Director{" +
                "filmography=" + filmography +
                ", style=" + style +
                '}';
    }
}
